cmd/metricsbridge/metricsbridge.go

```diff
@@ -233,6 +241,15 @@ func (c *config) runOnce(ctx context.Context) error {
 			for k, v := range sample.Metric {
 				if k != model.MetricNameLabel {
 					f.Dims[string(k)] = string(v)
+					//if there's an instance dimension try to lookup the hostname
+					if string(k) == "instance" {
+						hostname, present := hostnameMap[strings.Split(string(v), ":")[0]]
+						if present {
+							//only add the hostname dimension if the lookup succeeds
+							f.Dims["hostname"] = hostname
+							f.Dims["agentRole"] = strings.Split(hostname, "-")[0]
```

Comment:
this doesn't work - users can choose their compute agent pool name so you could have a hostname (e.g.) `my-compute-1234567890-000000`.  You could use the following function: if the first part of the hostname is master -> master; if it's infra -> infra; anything else -> compute

Reply:
s/agentRole/agentrole/ ?

Reply:
Changed capitalisation to "agentrole".

Changed role determination based on: "master","infra", everything else.